sklearn/metrics/_ranking.py

```diff
@@ -1487,20 +1495,27 @@ def _dcg_sample_scores(y_true, y_score, k=None, log_base=2, ignore_ties=False):
         Cumulative Gain (the DCG obtained for a perfect ranking), in order to
         have a score between 0 and 1.
     """
-    discount = 1 / (np.log(np.arange(y_true.shape[1]) + 2) / np.log(log_base))
+    xp, _, device = get_namespace_and_device(y_true, y_score)
+    max_float_dtype = _max_precision_float_dtype(xp, device)
+    log_base = xp.asarray(log_base, device=device, dtype=max_float_dtype)
+    discount = 1 / (
+        xp.log(xp.arange(y_true.shape[1], dtype=max_float_dtype, device=device) + 2)
```

Comment:
Just a question for my education, is this to make sure `arange` gives an array with `max_float_dtype` ?  Does that matter if `log_base` already has `max_float_dtype` ?